sql/core/src/main/scala/org/apache/spark/sql/execution/SubqueryBroadcastExec.scala

```diff
@@ -93,7 +93,7 @@ case class SubqueryBroadcastExec(
         val rows = if (broadcastRelation.keyIsUnique) {
           keyIter.toArray[InternalRow]
         } else {
-          keyIter.toSet[InternalRow].toArray
+          keyIter.toArray[InternalRow].distinct
```

Comment:
The disorder cause some bad case?

Reply:
I reverted this for now. Let's see how it gose.